src/map/script.c

```diff
@@ -18711,13 +18725,86 @@ BUILDIN_FUNC(waitingroom2bg_single)
 	if( (sd = cd->usersd[0]) == NULL )
 		return SCRIPT_CMD_SUCCESS;
 
-	if( bg_team_join(bg_id, sd) )
+	if( bg_team_join(bg_id, sd) && pc_setpos(sd, mapindex, x, y, CLR_TELEPORT) == SETPOS_OK)
 	{
-		pc_setpos(sd, mapindex, x, y, CLR_TELEPORT);
-		script_pushint(st,1);
+		script_pushint(st, true);
 	}
 	else
-		script_pushint(st,0);
+		script_pushint(st, false);
+
+	return SCRIPT_CMD_SUCCESS;
+}
+
+
+/// Creates an instance of battleground battle group.
+/// *bg_create("<map name>",<x>,<y>{,"<On Quit Event>","<On Death Event>"});
+/// @author [secretdataz]
+BUILDIN_FUNC(bg_create) {
+	const char *map_name, *ev = "", *dev = "";
+	int x, y, mapindex = 0;
+
+	map_name = script_getstr(st, 2);
+	if (strcmp(map_name, "-") != 0 && (mapindex = mapindex_name2id(map_name)) == 0)
+	{ // Invalid Map
+		script_pushint(st, 0);
+		return SCRIPT_CMD_SUCCESS;
+	}
+
+	x = script_getnum(st, 3);
+	y = script_getnum(st, 4);
+	if(script_hasdata(st, 5))
+		ev = script_getstr(st, 5); // Logout Event
+	if(script_hasdata(st, 6))
+		dev = script_getstr(st, 6); // Die Event
+
+	check_event(st, ev);
+	check_event(st, dev);
+
+	script_pushint(st, bg_create(mapindex, x, y, ev, dev));
+	return SCRIPT_CMD_SUCCESS;
+}
+
+/// Adds attached player or <char id> (if specified) to an existing 
+/// battleground group and warps it to the specified coordinates on
+/// the given map.
+/// bg_join(<battle group>,{"<map name>",<x>,<y>{,<char id>}});
+/// @author [secretdataz]
+BUILDIN_FUNC(bg_join) {
+	const char* map_name;
+	struct map_session_data *sd;
+	struct battleground_data *bg;
+	int x, y, bg_id, mapindex;
+
+	bg_id = script_getnum(st, 2);
+	if ((bg = bg_team_search(bg_id)) == NULL) {
+		script_pushint(st, false);
+		return SCRIPT_CMD_SUCCESS;
+	}
+	if (script_hasdata(st, 3)) {
+		map_name = script_getstr(st, 3);
+		if ((mapindex = mapindex_name2id(map_name)) == 0) {
+			script_pushint(st, false);
+			return SCRIPT_CMD_SUCCESS; // Invalid Map
+		}
+		x = script_getnum(st, 4);
+		y = script_getnum(st, 5);
+	} else {
+		mapindex = bg->mapindex;
+		x = bg->x;
+		y = bg->y;
+	}
+
+	if ((sd = script_charid2sd(6, sd)) == NULL) {
```

Comment:
Got this while compiling:

script.c: In function 'buildin_bg_join':
script.c:18801:10: warning: assignment makes pointer from integer without a cast [enabled by default]

Also, bg_join crashes the map server, probably it is caused by this.

This is the code that I've been using to test the new commands:
```
prontera,155,180,0	script	bg#bg_join	666,{
OnInit:
	waitingroom "BG Test",2;
	end;
}

bat_room,1,1,0		script	Flavius Init	-1,{
OnInit:
	bindatcmd "bgjoin","Flavius Init::OnBGJoin";
	bindatcmd "bgcreate","Flavius Init::OnBGCreate";
	bindatcmd "bgleave","Flavius Init::OnBGLeave";
	bindatcmd "bgdestroy","Flavius Init::OnBGDestroy";
	bindatcmd "bgwp","Flavius Init::OnBGwp";
	bindatcmd "bgchatjoin","Flavius Init::OnBGChatJoin";
	end;
OnBGJoin:
	bg_join($@FlaviusBG1_id1,"bat_b01",390,10,getcharid(0));
	end;
OnBGCreate:
	$@FlaviusBG1_id1 = bg_create("bat_b01",390,10);
	end;
OnBGLeave:
	bg_leave;
	end;
OnBGDestroy:
	if ($@FlaviusBG1_id1){
		bg_destroy ($@FlaviusBG1_id1);
	}
	end;
OnBGWp:
	bg_warp $@FlaviusBG1_id1,"bat_b01",390,10;
	end;
OnBGChatJoin:
	waitingroom2bg_single($@FlaviusBG1_id1,"bat_b01",390,10,"bg#bg_join");
	end;
}
```
bg_create successfully creates the BG group and attach the respawn coordinates to the battle group, I've tested this by waitingroom2bg_single since bg_join crashes map server.